drivers/sensor/lsm6dso/lsm6dso.c

```diff
@@ -766,28 +744,35 @@ static int lsm6dso_init_chip(const struct device *dev)
 
 	k_busy_wait(100);
 
-	if (lsm6dso_accel_set_fs_raw(dev,
-				     LSM6DSO_DEFAULT_ACCEL_FULLSCALE) < 0) {
-		LOG_DBG("failed to set accelerometer full-scale");
+	fs = cfg->accel_range;
+	LOG_DBG("accel range is %d", fs);
+	if (lsm6dso_accel_set_fs_raw(dev, fs) < 0) {
+		LOG_DBG("failed to set accelerometer full-scale %d", fs);
 		return -EIO;
 	}
-	lsm6dso->acc_gain = LSM6DSO_DEFAULT_ACCEL_SENSITIVITY;
+	lsm6dso->acc_gain = (lsm6dso_accel_fs_sens[fs] * GAIN_UNIT_XL);
 
-	lsm6dso->accel_freq = lsm6dso_odr_to_freq_val(CONFIG_LSM6DSO_ACCEL_ODR);
-	if (lsm6dso_accel_set_odr_raw(dev, CONFIG_LSM6DSO_ACCEL_ODR) < 0) {
-		LOG_DBG("failed to set accelerometer sampling rate");
+	odr = cfg->accel_odr;
+	LOG_DBG("accel odr is %d", odr);
+	lsm6dso->accel_freq = lsm6dso_odr_to_freq_val(odr);
+	if (lsm6dso_accel_set_odr_raw(dev, odr) < 0) {
+		LOG_DBG("failed to set accelerometer sampling rate %d", odr);
```

Comment:
This should be `LOG_ERR`. It's also a little confusing that you call it "accelerometer sampling rate" here, but "accel odr" two lines above.